scarb/src/core/workspace.rs

```diff
@@ -1,31 +1,52 @@
+use std::collections::BTreeMap;
 use std::fmt;
 
 use anyhow::Result;
-use camino::Utf8Path;
+use camino::{Utf8Path, Utf8PathBuf};
 
 use scarb_metadata::packages_filter::PackagesSource;
 
 use crate::compiler::Profile;
 use crate::core::config::Config;
 use crate::core::package::Package;
+use crate::core::PackageId;
 use crate::flock::RootFilesystem;
 use crate::MANIFEST_FILE_NAME;
 
-// TODO(#12): Support real workspaces.
 /// The core abstraction for working with a workspace of packages.
 ///
-/// **Note:** Currently only single-package workspaces are supported.
-///
 /// A workspace is often created very early on and then threaded through all other functions.
 /// It's typically through this object that the current package is loaded and/or learned about.
 pub struct Workspace<'c> {
     config: &'c Config,
-    package: Package,
+    members: BTreeMap<PackageId, Package>,
+    root_package: Option<PackageId>,
+    manifest_path: Utf8PathBuf,
```

Comment:
shouldn't we store manifest here?

Reply:
bump

Reply:
I feel like `core::manifest::Manifest` is really concerned with being the **package manifest**, defining targets, dependencies etc. We could create some `WorkspaceManifest` representation, although I do not think we need it **yet**. All the values defined for `.workspace = true` manifest inheritance can be merged when deserializing package manifests (like we do as of this pr), without loosing any functionality.

Reply:
Makes sense